prisma-fmt/src/code_actions/block.rs

```diff
@@ -0,0 +1,87 @@
+use std::collections::HashMap;
+
+use lsp_types::{CodeAction, CodeActionKind, CodeActionOrCommand, Diagnostic, Range, TextEdit, Url, WorkspaceEdit};
+use psl::{diagnostics::Span, parser_database::walkers::ModelWalker, schema_ast::ast::WithSpan};
+
+use super::CodeActionsContext;
+
+pub(super) fn create_missing_block(
+    actions: &mut Vec<CodeActionOrCommand>,
+    context: &CodeActionsContext<'_>,
+    model: ModelWalker<'_>,
+) {
+    let span_model = model.ast_model().span();
+    let diagnostics = context
+        .diagnostics_for_span_with_message(span_model, "is neither a built-in type, nor refers to another model,");
+
+    if diagnostics.is_empty() {
+        return;
+    }
+
+    let span = Span {
+        start: model.ast_model().span().start,
+        end: model.ast_model().span().end + 1, // * otherwise it's still not outside the closing brace
+        file_id: model.ast_model().span().file_id,
+    };
+
+    let range = super::range_after_span(context.initiating_file_source(), span);
+
+    diagnostics.iter().for_each(|diag| {
+        push_missing_block(
```

Comment:
Nice thing we can do for mongo specifically: don't suggest to create model if we are inside of a composite type (composite types can not reference models).
Not necessary though, so feel free to ignore.

Reply:
(If we don't add now, turn into issue)

Reply:
This is already only suggested from inside a model (& view).

```rs
/// code_actions.rs
// models AND views
    for model in validated_schema
        .db
        .walk_models_in_file(initiating_file_id)
        .chain(validated_schema.db.walk_views_in_file(initiating_file_id))
    {
        block::create_missing_block(&mut actions, &context, model);
```

I can add one for composite types to suggest further composite types. I don't think enums can hold _any_ references, if I'm not mistaken

Reply:
I had a false memory that `walk_models` also includes composites. It does not, you are right. 

>I can add one for composite types to suggest further composite types.

Let's do it. Would you prefer to do it now or later?

>I don't think enums can hold any references, if I'm not mistaken
Yes, I don't think there is anything more we can do for enums specifically

Reply:
Actually, that comment is out of date, `walk_models` no longer includes views and we have a separate views walker now apparently